--- lib/filter.c.orig	2009-05-08 17:46:43.000000000 +0200
+++ lib/filter.c	2009-06-18 14:06:18.000000000 +0200
@@ -68,28 +68,7 @@
     } u;
 };
 
-/* List of access_list. */
-struct access_list_list
-{
-  struct access_list *head;
-  struct access_list *tail;
-};
-
-/* Master structure of access_list. */
-struct access_master
-{
-  /* List of access_list which name is number. */
-  struct access_list_list num;
-
-  /* List of access_list which name is string. */
-  struct access_list_list str;
-
-  /* Hook function which is executed when new access_list is added. */
-  void (*add_hook) (struct access_list *);
-
-  /* Hook function which is executed when access_list is deleted. */
-  void (*delete_hook) (struct access_list *);
-};
+// struct definitions moved to header
 
 /* Static structure for IPv4 access_list's master. */
 static struct access_master access_master_ipv4 = 
