--- lib/filter.h.orig	2009-05-08 17:46:43.000000000 +0200
+++ lib/filter.h	2009-06-18 14:06:39.000000000 +0200
@@ -64,4 +64,27 @@
 extern struct access_list *access_list_lookup (afi_t, const char *);
 extern enum filter_type access_list_apply (struct access_list *, void *);
 
+/* List of access_list. */
+struct access_list_list
+{
+  struct access_list *head;
+  struct access_list *tail;
+};
+
+/* Master structure of access_list. */
+struct access_master
+{
+  /* List of access_list which name is number. */
+  struct access_list_list num;
+
+  /* List of access_list which name is string. */
+  struct access_list_list str;
+
+  /* Hook function which is executed when new access_list is added. */
+  void (*add_hook) (struct access_list *);
+
+  /* Hook function which is executed when access_list is deleted. */
+  void (*delete_hook) (struct access_list *);
+};
+
 #endif /* _ZEBRA_FILTER_H */
