--- lib/if.h.orig	2009-05-08 17:46:43.000000000 +0200
+++ lib/if.h	2009-06-18 14:10:52.000000000 +0200
@@ -304,8 +304,16 @@
 extern struct cmd_element no_interface_desc_cmd;
 extern struct cmd_element interface_cmd;
 extern struct cmd_element no_interface_cmd;
-extern struct cmd_element interface_pseudo_cmd;
-extern struct cmd_element no_interface_pseudo_cmd;
+// interface_pseudo_cmd not used
+// no_interface_pseudo_cmd not used
 extern struct cmd_element show_address_cmd;
 
+typedef int (*if_hook_func) (struct interface *);
+
+struct if_master
+{
+  if_hook_func if_new_hook;
+  if_hook_func if_delete_hook;
+};
+
 #endif /* _ZEBRA_IF_H */
