--- lib/if_rmap.c.orig	2009-05-08 17:46:43.000000000 +0200
+++ lib/if_rmap.c	2009-06-18 14:15:40.000000000 +0200
@@ -30,8 +30,9 @@
 struct hash *ifrmaphash;
 
 /* Hook functions. */
-static void (*if_rmap_add_hook) (struct if_rmap *) = NULL;
-static void (*if_rmap_delete_hook) (struct if_rmap *) = NULL;
+// prototype defined in header file
+if_rmap_hook_func if_rmap_add_hook = NULL;
+if_rmap_hook_func if_rmap_delete_hook = NULL;
 
 static struct if_rmap *
 if_rmap_new (void)
