--- lib/plist.h.orig	2009-05-08 17:46:43.000000000 +0200
+++ lib/plist.h	2009-06-18 14:39:12.000000000 +0200
@@ -80,4 +80,33 @@
 extern void prefix_bgp_orf_remove_all (char *);
 extern int prefix_bgp_show_prefix_list (struct vty *, afi_t, char *);
 
+/* List of struct prefix_list. */
+struct prefix_list_list
+{
+  struct prefix_list *head;
+  struct prefix_list *tail;
+};
+
+/* Master structure of prefix_list. */
+struct prefix_master
+{
+  /* List of prefix_list which name is number. */
+  struct prefix_list_list num;
+
+  /* List of prefix_list which name is string. */
+  struct prefix_list_list str;
+
+  /* Whether sequential number is used. */
+  int seqnum;
+
+  /* The latest update. */
+  struct prefix_list *recent;
+
+  /* Hook function which is executed when new prefix_list is added. */
+  void (*add_hook) (struct prefix_list *);
+
+  /* Hook function which is executed when prefix_list is deleted. */
+  void (*delete_hook) (struct prefix_list *);
+};
+
 #endif /* _QUAGGA_PLIST_H */
