--- lib/privs.c.orig	2009-05-08 17:46:43.000000000 +0200
+++ lib/privs.c	2009-06-18 14:41:45.000000000 +0200
@@ -63,18 +63,8 @@
 static zebra_privs_current_t zprivs_null_state = ZPRIVS_RAISED;
 
 /* internal privileges state */
-static struct _zprivs_t
-{
-#ifdef HAVE_CAPABILITIES
-  pstorage_t caps;		/* working storage        */
-  pset_t *syscaps_p;		/* system-type requested permitted caps    */
-  pset_t *syscaps_i;     	/* system-type requested inheritable caps  */
-#endif /* HAVE_CAPABILITIES */
-  uid_t zuid,                 /* uid to run as            */
-        zsuid;                /* saved uid                */
-  gid_t zgid;                 /* gid to run as            */
-  gid_t vtygrp;               /* gid for vty sockets      */
-} zprivs_state;
+// struct definition moved to header file
+struct _zprivs_t zprivs_state;
 
 /* externally exported but not directly accessed functions */
 #ifdef HAVE_CAPABILITIES
