--- lib/routemap.c.orig	2009-05-08 17:46:43.000000000 +0200
+++ lib/routemap.c	2009-06-18 12:32:33.000000000 +0200
@@ -35,33 +35,7 @@
 /* Vector for route set rules. */
 static vector route_set_vec;
 
-/* Route map rule. This rule has both `match' rule and `set' rule. */
-struct route_map_rule
-{
-  /* Rule type. */
-  struct route_map_rule_cmd *cmd;
-
-  /* For pretty printing. */
-  char *rule_str;
-
-  /* Pre-compiled match rule. */
-  void *value;
-
-  /* Linked list. */
-  struct route_map_rule *next;
-  struct route_map_rule *prev;
-};
-
-/* Making route map list. */
-struct route_map_list
-{
-  struct route_map *head;
-  struct route_map *tail;
-
-  void (*add_hook) (const char *);
-  void (*delete_hook) (const char *);
-  void (*event_hook) (route_map_event_t, const char *); 
-};
+// struct definitions moved to header
 
 /* Master list of route map. */
 static struct route_map_list route_map_master = { NULL, NULL, NULL, NULL };
