--- lib/sigevent.h.orig	2009-05-08 17:46:43.000000000 +0200
+++ lib/sigevent.h	2009-06-18 12:32:33.000000000 +0200
@@ -50,4 +50,14 @@
 /* check whether there are signals to handle, process any found */
 extern int quagga_sigevent_process (void);
 
+struct quagga_sigevent_master_t
+{
+  struct thread *t;
+
+  struct quagga_signal_t *signals; 
+  int sigc;
+  
+  volatile sig_atomic_t caught;
+};
+
 #endif /* _QUAGGA_SIGNAL_H */
