--- lib/zclient.c.orig	2009-05-08 17:46:43.000000000 +0200
+++ lib/zclient.c	2009-06-18 12:32:33.000000000 +0200
@@ -37,7 +37,7 @@
 enum event {ZCLIENT_SCHEDULE, ZCLIENT_READ, ZCLIENT_CONNECT};
 
 /* Prototype for event manager. */
-static void zclient_event (enum event, struct zclient *);
+static void zclient_event (enum event whatever, struct zclient *justadoxybugworkaround);
 
 extern struct thread_master *master;
 
